commons/src/main/java/eu/itesla_project/commons/tools/CommandLineTools.java

```diff
@@ -0,0 +1,138 @@
+        for (Map.Entry<String, Collection<Tool>> entry : toolsByTheme.entrySet()) {
+            String theme = entry.getKey();
+            List<Tool> tools = new ArrayList<>(entry.getValue());
+            Collections.sort(tools, Comparator.comparing(t -> t.getCommand().getName()));
+            usage.append(theme != null ? theme : "Others").append(":").append(System.lineSeparator());
+            for (Tool tool : tools) {
+                usage.append(String.format("   %-40s %s", tool.getCommand().getName(), tool.getCommand().getDescription())).append(System.lineSeparator());
+            }
+            usage.append(System.lineSeparator());
+        }
+
+        err.print(usage);
+        return COMMAND_NOT_FOUND_STATUS;
+    }
+
+    private static Options getOptionsWithHelp(Options options) {
+        Options optionsWithHelp = new Options();
+        options.getOptions().forEach(optionsWithHelp::addOption);
+        optionsWithHelp.addOption(Option.builder()
+                .longOpt("help")
+                .desc("display the help and quit")
+                .build());
+        return optionsWithHelp;
+    }
+
+    private void printCommandUsage(Command command, PrintStream err) {
+        HelpFormatter formatter = new HelpFormatter();
+        PrintWriter writer = new PrintWriter(err);
+        formatter.printHelp(writer,
+                            80,
+                            TOOL_NAME + " " + command.getName(),
+                            "", // header
+                            getOptionsWithHelp(command.getOptions()),
+                            formatter.getLeftPadding(),
+                            formatter.getDescPadding(),
+                            System.lineSeparator() + Objects.toString(command.getUsageFooter(), ""),
+                            true);
+        writer.flush();
+    }
+
+    private Tool findTool(String commandName) {
+        for (Tool tool : tools) {
+            if (tool.getCommand().getName().equals(commandName)) {
+                return tool;
+            }
+        }
+        return null;
+    }
+
+    public int run(String[] args, ToolRunningContext context) throws IOException {
+        Objects.requireNonNull(args);
+        Objects.requireNonNull(context);
+
+        if (args.length < 1) {
+            return printUsage(context.getErr());
+        }
+
+        Tool tool = findTool(args[0]);
+        if (tool == null) {
+            return printUsage(context.getErr());
+        }
+
+        try {
+            CommandLineParser parser = new DefaultParser();
+            if (Arrays.asList(args).contains("--help")) {
+                printCommandUsage(tool.getCommand(), context.getErr());
+            } else {
+                CommandLine line = parser.parse(getOptionsWithHelp(tool.getCommand().getOptions()), Arrays.copyOfRange(args, 1, args.length));
+                tool.run(line, context);
+            }
+            return COMMAND_OK_STATUS;
+        } catch (ParseException e) {
```

Comment:
If tool.run() throws a ParseException, the status code should be "EXECUTION_ERROR_STATUS" instead of "INVALID_COMMAND_STATUS". The ParseException should be catched earlier